### PR TITLE
Releasing process involving artifactory

### PR DESCRIPTION
**Pull request by nbirkenshaw-mux**
This is the release process that involves pushing AARs to artifactory for easier consumption. There is an example in demo/build.gradle (commented out in the dependencies section) which shows how you would import it and the appropriate ExoPlayer.

All builds from this get pushed to the default-maven-local repo in our artifactory, which enables us to then reference those aars elsewhere for testing.

If you do a merge to master with a version commit message (i.e. it starts with v5.6.7) then that will, on successful completion of the tests, prompt you on buildkite to deploy the build to default-maven-release-local which is intended to have public read setup once this whole thing is complete.

No claims here that this is particularly clean/nice. I think we're up against the edge of sanity with our gradle config for all the variants!

This is a cleaned up version of the work on nb-jfrog. The pre-existing JavaDoc work on the jfrog branches will be turned into separate PRs.

### CODE REVIEW

**Review by skidder — APPROVED** [review]

**Review by tomkordic — APPROVED** [review]

File: MuxExoPlayer/build.gradle [hunk 1]
```diff
@@ -28,7 +30,7 @@ android {
         minSdkVersion project.ext.minSdkVersion
         targetSdkVersion project.ext.targetSdkVersion
         versionCode 26
-        versionName "2.4.15"
+        versionName project.ext.versionName
```
Comment (jsanford8) [inline MuxExoPlayer/build.gradle:33]: Just one quick question - this is set correctly elsewhere as 2.4.15, right?

**Review by jsanford8 — APPROVED** [review]
One small check to make, but assuming the version is set elsewhere, we're good to go

File: MuxExoPlayer/build.gradle (hunk 1 above)
Comment (nbirkenshaw-mux) [reply MuxExoPlayer/build.gradle:33]: It'll base the version name on the git commit message for PRs merged to master. Nowhere is this now set to 2.4.15. If the commit message starts with a vX.X.X pattern then it'll assume you want to make a public release build with that version

In the event that it's not a merge to master (i.e. it's almost anything else that would trigger a build) you will still get a build on our (to be) private maven repo name "dev-[branch]-[hash]" for testing purposes.

Comment (jsanford8) [reply MuxExoPlayer/build.gradle:33]: oh even better, works for me!

**Pull request merged**